pkg/rollout/rollout.go

```diff
@@ -98,14 +103,35 @@ func (r *Rollout) UpdateService(svc *run.Service) (*run.Service, error) {
 	}
 	r.log.Debugf("%q is the stable revision", stable)
 
-	candidate := DetectCandidateRevisionName(svc, stable)
+	candidate, isNewCandidate := DetectCandidateRevisionName(svc, stable)
 	if candidate == "" {
 		r.log.Info("could not determine candidate revision")
 		return nil, nil
 	}
-	r.log.Debugf("%q is the candidate revision", candidate)
 
-	svc = r.SplitTraffic(svc, stable, candidate)
+	var diagnosis *health.Diagnosis
```

Comment:
if you look at `if isNewCandidate` you can define diagnosis later

Reply:
I refactored this piece to make the possible paths in the code clearer